example/provider-local/garden/base/kustomization.yaml

```diff
@@ -19,6 +19,9 @@ patches:
     kind: ControllerRegistration
     name: provider-local
   patch: |
+    - op: add
+      path: /metadata/annotations/security.gardener.cloud~1pod-security-enforce
+      value: privileged
```

Comment:
Shouldn't we adapt https://github.com/gardener/gardener/blob/master/hack/generate-controller-registration.sh to let this annotation become part of the generated `ControllerRegistration`? Otherwise, how would human operators know what/whether to set it when deploying the `ControllerRegistration` into their garden clusters?

Reply:
We would want to add `security.gardener.cloud/pod-security-enforce: privileged` to `ControllerRegistration`s only if their `extension` `namespace` has components that do no comply with `baseline`.  Ideally we would place this annotation to default to `baseline` but then we run the risk of having problems with unknown not compliant pods.
 After releasing this annotation and validating that all components are working correctly on clusters across our landscapes by  configuring the admission controller to enforce `baseline` we can adapt https://github.com/gardener/gardener/blob/master/hack/generate-controller-registration.sh to set `baseline` for generated `ControllerRegistration`. If any were not compliant with `baseline` we will set them to `privileged`

Reply:
I have several questions:

> We would want to add `security.gardener.cloud/pod-security-enforce: privileged` to `ControllerRegistration`s only if their `extension` `namespace` has components that do no comply with `baseline`. Ideally we would place this annotation to default to `baseline` but then we run the risk of having problems with unknown not compliant pods.

I don't understand the target picture. I thought to have understood that each extension publishes in its `ControllerRegistration` via this annotation which pod security enforcement it wants. Default is `baseline`, but there could be cases where it's overwritten with `privileged`. Then I thought that `gardenlet` only annotates extension namespaces when the `ControllerRegistration` has this annotation in the first place, and if it's not present then it does nothing.
Hence, I suggested that all `ControllerRegistration`s need to publish this annotation (otherwise, human operators cannot know what to put there).

Your answer does not match my understanding, so apparently it's wrong. What is the target picture then?

> After releasing this annotation and validating that all components are working correctly on clusters across our landscapes

What about extensions that we don't know about or have not registered in "our landscapes"?

> by configuring the admission controller to enforce `baseline`

I think this was discussed above already: What about seed clusters where we don't have access to change the admission config of the kube-apiserver?

> If any were not compliant with `baseline` we will set them to `privileged`

How? You mean making it configurable in the `generate-controller-registration.sh` script?

Reply:
As discussed in Slack, I made the annotation configurable in the `generate-controller-registration.sh` script. The `ControllerRegistration` annotation now defaults to `baseline` and can be configured via the `-e, --pod-security-enforce[=pod-security-standard]` option. The option has been added to the generate command of `provider-local` to set `privileged` policy.
The change has been added to the release notes as breaking, since we cannot guarantee that all extension will comply with `baseline`. If a given extension cannot comply with `baseline` the extension's owners would need to add the `-e, --pod-security-enforce[=pod-security-standard]` option set to `privileged` to the generate command.